src/hooks/useAutoHeight.ts

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+
+export function useAutoHeight() {
+  const [containerRef, setContainerRef] = React.useState<
+    HTMLElement | undefined
+  >();
+  const refCallback = (node: HTMLElement) => {
```

Comment:
Using `HTMLElement` here instead of `JSX.Element` because `getBoundingClientRect()`, `clientHeight`, and `paddingBottom` are not on `JSX.Element` and TS didn't like it when I tried to cast between the two types. If anyone has a better solution for this please let me know.